x-pack/plugins/task_manager/server/routes/health.ts

```diff
@@ -0,0 +1,169 @@
+  [ServiceStatusLevels.unavailable.toString()]: 'Task Manager is unavailable',
+};
+
+export function healthRoute(
+  router: IRouter,
+  monitoringStats$: Observable<MonitoringStats>,
+  logger: Logger,
+  taskManagerId: string,
+  config: TaskManagerConfig
+): Observable<ServiceStatus> {
+  // if "hot" health stats are any more stale than monitored_stats_required_freshness (pollInterval +1s buffer by default)
+  // consider the system unhealthy
+  const requiredHotStatsFreshness: number = config.monitored_stats_required_freshness;
+
+  // if "cold" health stats are any more stale than the configured refresh (+ a buffer), consider the system unhealthy
+  const requiredColdStatsFreshness: number = config.monitored_aggregated_stats_refresh_rate * 1.5;
+
+  function calculateStatus(monitoredStats: MonitoringStats): MonitoredHealth {
+    const now = Date.now();
+    const timestamp = new Date(now).toISOString();
+    const summarizedStats = summarizeMonitoringStats(monitoredStats, config);
+
+    /**
+     * If the monitored stats aren't fresh, return a red status
+     */
+    const healthStatus =
+      hasStatus(summarizedStats.stats, HealthStatus.Error) ||
+      hasExpiredHotTimestamps(summarizedStats, now, requiredHotStatsFreshness) ||
+      hasExpiredColdTimestamps(summarizedStats, now, requiredColdStatsFreshness)
+        ? HealthStatus.Error
+        : hasStatus(summarizedStats.stats, HealthStatus.Warning)
+        ? HealthStatus.Warning
+        : HealthStatus.OK;
+    return { id: taskManagerId, timestamp, status: healthStatus, ...summarizedStats };
+  }
+
+  const serviceStatus$: Subject<ServiceStatus> = new Subject<ServiceStatus>();
+
+  /* keep track of last health summary, as we'll return that to the next call to _health */
+  let lastMonitoredStats: MonitoringStats | null = null;
+
+  /* Log Task Manager stats as a Debug log line at a fixed interval */
+  monitoringStats$
+    .pipe(
+      throttleTime(requiredHotStatsFreshness),
+      tap((stats) => {
+        lastMonitoredStats = stats;
+      }),
+      // Only calculate the summerized stats (calculates all runnign averages and evaluates state)
+      // when needed by throttling down to the requiredHotStatsFreshness
+      map((stats) => withServiceStatus(calculateStatus(stats)))
+    )
+    .subscribe(([monitoredHealth, serviceStatus]) => {
+      serviceStatus$.next(serviceStatus);
+      logger.debug(JSON.stringify(monitoredHealth));
+    });
+
+  router.get(
+    {
+      path: '/api/task_manager/_health',
+      validate: false,
+    },
+    async function (
+      context: RequestHandlerContext,
+      req: KibanaRequest<unknown, unknown, unknown>,
+      res: KibanaResponseFactory
+    ): Promise<IKibanaResponse> {
+      return res.ok({
+        body: lastMonitoredStats
+          ? calculateStatus(lastMonitoredStats)
+          : { id: taskManagerId, timestamp: new Date().toISOString(), status: HealthStatus.Error },
```

Comment:
This is really more of a "NO DATA" condition, rather than an error, right?  Probably worth indicating that _somehow_, not sure it's worthy of a new `status` if instead it could just be in a message somehow.

Reply:
I considered it, but arguably it *could* also be an error if you consider that the goal of monitoring is to confirm that TM is working and having no data here indicates that TM is *not* working.

My thinking was that when a Kibana starts up it goes from Error to OK, and that makes sense to me fro ma monitoring perspective...
I'd opt to keep it this way until (unless) a customer complains :)